ui/src/components/Exchange/CreateMarket.tsx

```diff
@@ -34,32 +39,111 @@ const CreateMarket: React.FC<{}> = () => {
     const operator = useOperator();
 
     const allTokens: TokenInfo[] = useContractQuery(Token);
+    const allDerivatives: DerivativeInfo[] = useContractQuery(Derivative);
     const registeredCCPs = useContractQuery(RegisteredCCP, AS_PUBLIC);
     const quantityPrecision = Number(baseToken?.contractData.quantityPrecision) || 0
 
-    const handleIdPairSubmit = async () => {
-        if (!baseToken || !quoteToken) {
-            throw new Error('Tokens not selected');
+    const allTokenOptions = allTokens.map(tk => {
+            return {
+                key: tk.contractId,
+                text: `${tk.contractData.id.label} - Spot`,
+                value: tk.contractId
+            }
+        });
+
+    const allDerivativeOptions = allDerivatives.map(dr => {
+        return {
+            key: dr.contractId,
+            text: `${dr.contractData.id.label} - Derivative`,
+            value: dr.contractId
         }
+    });
+    const tokenMap = new Map(allTokens.map(tk => [String(tk.contractId), tk.contractData.id]));
+    const derivativeMap = new Map(allDerivatives.map(dr => [String(dr.contractId), dr.contractData.id]));
+    const allMap = new Map([...Array.from(tokenMap.entries()), ...Array.from(derivativeMap.entries())]);
+    const allOptions = [...allTokenOptions, ...allDerivativeOptions]
```

Comment:
There might be a better way to set this up...since the option list has both derivatives and tokens, there needed to be a way of differentiating if the selected option was a derivative or a token and get the corresponding `Id` from that.

Reply:
What about adding another field to the map, `assetType: AssetType` and adding it in the tokenMap/derivativeMap

Then, when it's all merged into `allMap`, you can later do (in the `getType()` function), 

```
const asset = allMap.get(val);
if (!asset) {
    throw new Error('Option not found')
}
return asset.assetType
```

Reply:
Nice, that works great!